v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java

```diff
@@ -120,20 +120,6 @@ public interface SourceDbToSpannerOptions extends CommonTemplateOptions {
 
   void setPassword(String password);
 
-  @TemplateParameter.Text(
-      order = 9,
-      optional = true,
```

Comment:
If this was optional, why delete rather than add a default ? There is a value in keeping this - sometimes a non-primary key column could help in partitioning.
Also - the numPartitions is having a default - we should either delete both or keep both,IMO.

Reply:
# PartitionColumns
The Reader allows a map of partition columns to be configured. For the pipeline we are leaning on starting with less configurability (for simplicity). 

A bit more detailed:
Earlier to this PR, we had both tables and partition columns configured in the pipeline options as a list (of equal size). Now we let the tables be empty and auto discover tables. This changed the format of configuring partition column to a map<TableName, Column> (as the reader can autodiscover all tables and let the upper layer configure few partition columns that really need an override). At this point, we had an option to accept the map in the pipeline configuration like "table:column, table:column...", but we decided to keep it simple for now and remove the external configuration for partition column to start with.
We can add it back if we need that in a real situation (it's just at the level of options and the underlying layer of reader is still configurable).

# Num Partitions.
NumPartitions is needed till we have more detailed splitting logic or enhancements in JDBCIO. JdbCIO just timesout taking counts and throws exception if numPartitions is not configured and dataset is large. Eventually we will remove this too.